sklearn/tree/_utils.pyx

```diff
@@ -25,17 +24,19 @@ np.import_array()
 # Helper functions
 # =============================================================================
 
-cdef realloc_ptr safe_realloc(realloc_ptr* p, size_t nelems) except *:
+cdef realloc_ptr safe_realloc(realloc_ptr* p, size_t nelems) nogil except *:
```

Comment:
This should be `except NULL` as `NULL` is only returned in case of `MemoryError` which is the only exception raised by this function.